gpMgmt/test/behave/mgmt_utils/gpperfmon.feature

```diff
@@ -101,6 +101,32 @@ Feature: gpperfmon
         """
         Then wait until the results from boolean sql "SELECT count(*) > 0 FROM queries_history WHERE query_text = 'SELECT pg_sleep(80)'" is "true"
 
+    @gpperfmon_query_history
+    Scenario Outline: gpperfmon does not log nested statements with log_min_messages < debug4
+        Given gpperfmon is configured and running in qamode
+        When the user truncates "queries_history" tables in "gpperfmon"
+        When below sql is executed in "gptest" db
+        """
+        CREATE OR REPLACE FUNCTION test_sleep() RETURNS SETOF INT AS $$BEGIN
+            RETURN QUERY SELECT 1 FROM pg_sleep(30);
+        END$$ LANGUAGE plpgsql;
+        """
+        When below sql is executed in "gptest" db line by line as a transaction
```

Comment:
why not use `When below sql is executed in "gptest" db` and explicit `BEGIN;` and `COMMIT;` blocks?

Reply:
`When below sql is executed in "gptest" db` executes the whole text as one command, so both `DECLARE test_cursor` and `SELECT test_sleep()` end up in `query_text` for both queries. This doesn't allow checking for them separately.

Reply:
> `When below sql is executed in "gptest" db` executes the whole text as one command, so both `DECLARE test_cursor` and `SELECT test_sleep()` end up in `query_text` for both queries.

What is the expected behavior of `gpperfmon` in this case?

Reply:
> this is a separate bug with multi-statement queries handling (possibly even in the python GPDB driver), not in gpperfmon.

how about
```bash
psql -1 <<EOF
SET log_min_messages = "<log_level>";
DECLARE test_cursor CURSOR FOR SELECT * FROM generate_series(1,100);
FETCH FORWARD 10 FROM test_cursor;
SELECT test_sleep();
EOF
```

Reply:
This would probably work, at least the interactive mode handles this correctly. I will check this

Reply:
Like that?